File: src/main/java/me/leodev/trollingfreedomreborn/trolls/Beds/BedExplosion.java
package me.leodev.trollingfreedomreborn.trolls.Beds;

import org.bukkit.World;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerBedEnterEvent;

import java.util.ArrayList;

public class BedExplosion implements Listener {

    public static ArrayList<String> Explode1 = new ArrayList();


    public void BedExplosion(Player p) {
        String p2 = p.getName();
        Explode1.add(p.getName());
    }

    public void UnBedExplosion(Player p) {
        String p2 = p.getName();
        Explode1.remove(p.getName());
    }

    @EventHandler
    public void onSex(PlayerBedEnterEvent event) {
        Player p = event.getPlayer();
        if (Explode1.contains(p.getName())) {
            World w = p.getWorld();
            w.createExplosion(event.getBed().getLocation(), 3, true);
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/Beds/BedExplosion.kt
package com.leomadrassi.trollingfreedomreborn.trolls.Beds

import org.bukkit.World
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerBedEnterEvent

class BedExplosion : Listener {
    companion object {
        val Explode1 = mutableListOf<String>()
    }

    fun BedExplosion(p: Player) {
        Explode1.add(p.name)
    }

    fun UnBedExplosion(p: Player) {
        Explode1.remove(p.name)
    }

    @EventHandler
    fun onSex(event: PlayerBedEnterEvent) {
        val p = event.player
        if (Explode1.contains(p.name)) {
            val w = p.world
            w.createExplosion(event.bed.location, 3.0f, true)
        }
    }
}
